ibis/bigquery/compiler.py

```diff
@@ -586,6 +586,19 @@ def compiles_floor(t, e):
     return 'CAST(FLOOR({}) AS {})'.format(t.translate(arg), bigquery_type)
 
 
+@compiles(ops.CMSMedian)
+def compiles_approx(translator, expr):
+    arg = expr.op().args[0]
+    where = expr.op().args[1]
+
+    if where is not None:
+        arg = where.ifelse(arg, ibis.NA)
+
+    return 'APPROX_QUANTILES({}, 2)[OFFSET(1)]'.format(
```

Comment:
`APPROX_QUANTILES` returns an array. By setting the number of quantiles to 2 it returns an array of  [min, median, max], so the offset is required to pull out the median.

Reply:
Is there any reason to use [`SAFE_OFFSET`](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#safe_offset-and-safe_ordinal) here?